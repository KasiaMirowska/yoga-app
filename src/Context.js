import React from 'react';
import APIFlowCalls from './services/API_Flow_service';


const YogaContext = React.createContext({
    currentFlowId: null,
    currentFlow: {},
    poses: [],
    flows: [],
    openPoseCard: {},
    setPosesList: () => { },
    setFlowsList: () => { },
    setOpenPoseCard: () => { },
    setAttributesIntoOpenCard: () => {},
    setCurrentNewFlow: () => { },
    setCurrentFlowId: () => { },
    setCurrentFlow: () => { },
    truncateCurrentFlow: () => { },
    addFlow: () => { },
    enterFlow: () => { },
    deletePoseFromFlow: () => {},
    poseAttributes: [],
})

export default YogaContext;


export class YogaContextProvider extends React.Component {
    constructor() {
        super()
        this.state = {
            currentFlowId: null,
            currentFlow: {
                id: null,
                title: '',
                assignedPoses: [],
                warmUp: [],
                midFlow: [],
                peakPose: [],
                breakPoses: [],
                afterPeak: [],
            },
            openPoseCard: {
                id: null,
                name_eng: '',
                name_san: '',
                benefits: '',
                pose_type: '',
                pose_level: '',
                img: '',
                video: '',
                attributesList: '',
                notes: '',

            },
            poses: [],
            flows: [],
            poseAttributes: [],
            error: null,
        }
    }


    setError = (res) => {
        this.setState({
            error: res,
        })
    }
    enterFlow = (flow) => {
        this.setState({
            currentFlowId: flow.id,
            currentFlow: {
                id: flow.id,
                title: flow.title,
                assignedPoses: [],
                peakPose: [],
                warmUp: [],
                midFlow: [],
                breakPoses: [],
                afterPeak: [],
            }
        })
    }
    

    setCurrentFlow = (flow) => {
        this.setState({
            currentFlowId: flow.id,
            currentFlow: flow,
        })
    }
    
    setPosesList = (data) => {
        this.setState({
            poses: data,
        })
    }

    setFlowsList = (data) => {
        this.setState({
            flows: data,
        })
    }

    setOpenPoseCard = (data) => {
         this.setState({
            openPoseCard: {
                id: data.id,
                name_eng: data.name_eng,
                name_san: data.name_san,
                benefits: data.benefits,
                pose_type: data.pose_type,
                pose_level: data.pose_level,
                img: data.img,
                video: data.video,
                attributesList: data.attributesList,
                notes: data.notes,
            }
        })
    }


    deletePoseFromFlow = (id) => {
        const newAssignedPoses = this.state.currentFlow.assignedPoses.map(posesArr => {
            return posesArr.filter(pose => pose !== id)
        })
        console.log(newAssignedPoses)
        this.setState({
            currentFlow: {
                ...this.state.currentFlow,
                assignedPoses: newAssignedPoses
            }
        })

        const poseToDelete = id;
        const flowAimed = this.state.currentFlowId;
        APIFlowCalls.deletePoseFromFlow(poseToDelete, flowAimed)
            .then((res) => {
                console.log('deleting POSE FROM FLOW')
            })
            .catch(err => {
                console.log(err)
                this.setState({
                    error: err,
                })
            })
       
    }
  


    render() {
        const contextValue = {

            openPoseCard: this.state.openPoseCard,
            currentFlowId: this.state.currentFlowId,
            currentFlow: this.state.currentFlow,
            poses: this.state.poses,
            flows: this.state.flows,
            setError: this.setError,
            truncateCurrentFlow: this.truncateCurrentFlow,
            setCurrentFlow: this.setCurrentFlow,
            setCurrentFlowId: this.setCurrentFlowId,
            setPosesList: this.setPosesList,
            setFlowsList: this.setFlowsList,
            setOpenPoseCard: this.setOpenPoseCard,
            setAttributesIntoOpenCard: this.setAttributesIntoOpenCard,
            setCurrentNewFlow: this.setCurrentNewFlow,
            addFlow: this.addFlow,
            enterFlow: this.enterFlow,
            deletePoseFromFlow: this.deletePoseFromFlow,
            poseAttributes: this.state.poseAttributes,
        }

        return (
            <YogaContext.Provider value={contextValue} >
                {this.props.children}
            </YogaContext.Provider >
        )
    }
}