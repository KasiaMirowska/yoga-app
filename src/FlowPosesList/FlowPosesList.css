
.flow-poses-container {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around; 
    /* width: 80%; */
    margin: auto;
    padding-bottom: 20px;
}



@media (min-width: 600px) {
    .flow-poses-container {
        display: flex;
        flex-wrap: wrap;
        justify-content: space-around; 
        width: 80%;
        margin: auto;
        padding-bottom: 20px;
    }
  }  