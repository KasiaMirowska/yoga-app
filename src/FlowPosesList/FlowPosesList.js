import React from 'react';
import YogaContext from '../Context';
import FlowItem from '../FlowItem/FlowItem';
import APIFlowCalls from '../services/API_Flow_service';

export default class FlowPosesList extends React.Component {
    static contextType = YogaContext;
    

    componentDidMount = () => {
        const flowId = this.context.currentFlowId;
        
        APIFlowCalls.getAllPosesInFlow(flowId)
            .then(data => {
                this.context.setCurrentFlow(data);
            })
    }
    
    render() {
        let currentFlowPosesIds = this.context.currentFlow.assignedPoses;
        const { poses } = this.context;
    
        let orderedIds = currentFlowPosesIds.map(element => element.map(id => {
            return poses.find(pose => pose.id === id)
        }))
        
        
        
        const flowPoses = orderedIds.map(element => element.map((pose, index) => {
            return < FlowItem
                key={index}
                poseId={pose.id}
                img={pose.img}
                flowId={this.context.currentFlow.id}
            />
        }))


        return (
            <div>
                {flowPoses}
            </div >
        )
    }
}