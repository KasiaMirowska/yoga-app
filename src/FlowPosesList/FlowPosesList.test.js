import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import FlowPosesList from './FlowPosesList';
import {BrowserRouter as BR} from 'react-router-dom';
import { YogaContextProvider as YCP } from '../Context';

describe('FlowPosesList component', () => {
    const props = {match: {params: {path: '/flow'}}}
    
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><YCP><FlowPosesList {...props} /></YCP></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
    it('renders UI as expected', () => {
        const loginPage = renderer.create(<BR><YCP><FlowPosesList {...props}/></YCP></BR>);
        expect(loginPage.toJSON()).toMatchSnapshot();
    })

})