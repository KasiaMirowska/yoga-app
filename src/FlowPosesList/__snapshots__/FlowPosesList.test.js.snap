// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowPosesList component renders UI as expected 1`] = `
<div>
  <div
    className="error"
  />
  <ul
    className="flow-poses-container"
  />
</div>
`;
